src/libraries/System.Text.Json/tests/NewtonsoftTests/JsonSerializerTests.cs

```diff
@@ -131,6 +131,15 @@ public void PreserveReferencesCallbackTest()
             p2.Spouse = p1;
             Assert.Throws<JsonException> (() => JsonSerializer.Serialize(p1));
         }
+
+        [Fact]
```

Comment:
Add tests for this in a new file "src\libraries\System.Text.Json\tests\Serialization\JsonDocumentTests.cs". Please include tests for serialization and invalid cases where we throw exceptions. Tests can be modeled after [`JsonElementTests`](https://github.com/dotnet/runtime/blob/7955318c2ccc79ca4412e21ed4f1cb1f06faf646/src/libraries/System.Text.Json/tests/Serialization/JsonElementTests.cs#L11).

Reply:
Thanks for the guidance - have added these tests.